### PR TITLE
refactor(services): merging ui and feature libs under a same one

### PR DESCRIPTION
**Pull request by bdebon**
# What does this PR do?

> Link to the JIRA ticket

Put description here

> Screenshot of the feature

Put screenshot following the template `before` => `after` here

---

### Global

- [ ] This PR does not introduce any breaking change
- [ ] This PR introduces breaking change(s) and has been labeled as such
- [ ] I have found someone to review this PR and pinged him

### Store

- [ ] This PR introduces new store changes

### NX

- [ ] I have run the dep-graph locally and made sure the tree was clean i.e no circular dependencies
- [ ] I have followed the library pattern i.e `feature`, `ui`, `data`, `utils`

### Clean Code

- [ ] I made sure the code is type safe (no any)
- [ ] I have included a feature flag on my feature, if applicable

### CODE REVIEW

**Comment by evoxmusic** [comment]
A preview environment was automatically created via Qovery.  
Click on the link below to follow its deployment and use it.  
:point_right: [[PR] staging - refactor(services): merging ui and feature libs under a same one - 2022-06-16T09:58:09Z](https://console.qovery.com/platform/organization/141c07c8-0dd9-4623-983b-3fdd61867255/projects/977bd069-20f7-4b24-bd95-28b44174cf43/environments/8d77a7b1-9d7a-483e-bb58-621c58c4e483/applications)

File: libs/pages/services/src/lib/feature/deployments/deployments-feature.tsx [hunk 1]
```diff
@@ -1,22 +1,20 @@
...
-import { DeploymentsPage } from '@console/pages/services/ui'
 import { useEffect } from 'react'
 import { useDispatch, useSelector } from 'react-redux'
 import { useParams } from 'react-router'
 import { ServicesEnum } from '@console/shared/enums'
-import { DeploymentHistoryApplication } from 'qovery-typescript-axios'
+import DeploymentsPage from '../../ui/deployments-page/deployments-page'
 
-export function Deployments() {
+export function DeploymentsFeature() {
```
Comment (bdebon) [inline libs/pages/services/src/lib/feature/deployments/deployments-feature.tsx:17]: Here, Deployments becomes DeploymentsFeature

File: libs/pages/services/src/lib/feature/deployments/deployments-feature.tsx [hunk 2]
```diff
@@ -70,7 +68,7 @@ export function Deployments() {
   }, [dispatch, environmentId, projectId, environment])
 
   return (
-    <DeploymentsPage
+    <Deployments
```
Comment (bdebon) [inline libs/pages/services/src/lib/feature/deployments/deployments-feature.tsx:71]: And `DeploymentsPage` becomes `Deployments`

File: libs/pages/services/src/lib/feature/general/general-feature.tsx [hunk 3]
```diff
@@ -13,12 +13,12 @@ import {
...
 } from '@console/domains/application'
 import { AppDispatch, RootState } from '@console/store/data'
-import { GeneralPage } from '@console/pages/services/ui'
 import { BaseLink, StatusMenuActions } from '@console/shared/ui'
 import { selectEnvironmentById } from '@console/domains/environment'
 import { databasesLoadingStatus, fetchDatabasesStatus, selectDatabasesEntitiesByEnvId } from '@console/domains/database'
+import { General } from '../../ui/general/general'
 
-export function General() {
+export function GeneralFeature() {
```
Comment (bdebon) [inline libs/pages/services/src/lib/feature/general/general-feature.tsx:21]: Same for General -> GeneralFeature

File: libs/pages/services/src/lib/feature/general/general-feature.tsx [hunk 4]
```diff
@@ -81,7 +81,7 @@ export function General() {
     databasesByEnv.length === 0
 
   return (
-    <GeneralPage
+    <General
```
Comment (bdebon) [inline libs/pages/services/src/lib/feature/general/general-feature.tsx:84]: Rendering `General` and not GeneralPage

**Review by bdebon — COMMENTED** [review]
I highlighted 4 places so you can see what I propose. Things that are defined in `feature` folder are prefixed with `Feature` so that with a single glimpse of an eye, we can see if we will find some logic in the component and if we can add some logic. Everything that does not have a prefix Feature is supposed to be an UI page that can't have logic.

I removed the notion of `Page` inside the lib. We have one Page per lib, the one we will call in the general router. Everything about subroute will call `Features`

Are you okay with these changes and rules?

File: libs/pages/services/src/lib/router/router.tsx [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import { Route, SERVICES_DEPLOYMENTS_URL, SERVICES_GENERAL_URL } from '@console/shared/router'
+import GeneralFeature from '../feature/general/general-feature'
+import DeploymentsFeature from '../feature/deployments/deployments-feature'
+
+export const ROUTER_SERVICES: Route[] = [
+  {
+    path: SERVICES_GENERAL_URL,
+    component: <GeneralFeature />,
```
Comment (bdebon) [inline libs/pages/services/src/lib/router/router.tsx:8]: Here in the Subrouting of this page, we don't call any Pages, we call Feature components. Ok?

File: libs/pages/services/src/lib/feature/deployments/deployments-feature.tsx (hunk 1 above)
Comment (RemiBonnet) [reply libs/pages/services/src/lib/feature/deployments/deployments-feature.tsx:17]: It's necessary to have `deployments-feature.tsx` ? we can have directly `deployments.tsx`?

File: libs/pages/services/src/lib/feature/deployments/deployments-feature.tsx (hunk 2 above)
Comment (RemiBonnet) [reply libs/pages/services/src/lib/feature/deployments/deployments-feature.tsx:71]: same here

**Pull request merged**